src/components/Homepage.tsx

```diff
@@ -11,13 +12,182 @@ import { PostType } from "../lib/types";
 
 const Homepage = () => {
   const navigateTo = useNavigate();
+  const [search, updateSearch] = useState("");
   const [query, updateQuery] = useState("");
+  const [selected, setSelected] = useState(false);
 
   const submitHandler: React.FormEventHandler<HTMLFormElement> = (event) => {
     event.preventDefault();
 
-    navigateTo(`/facebook?q=${query}`);
+    if (selected) navigateTo(`/facebook?q=${query}`);
+    else navigateTo(`/facebook?q=${search}`);
   };
+  const [name, updateName] = useState("");
+  const [unix, updateUnix] = useState("");
+  const [country, updateCountry] = useState("");
+  const [state, updateState] = useState("");
+  const [city, updateCity] = useState("");
+  const [button1, setButton1] = useState(false);
+  const [button2, setButton2] = useState(false);
+  const [button3, setButton3] = useState(false);
+  const [button4, setButton4] = useState(false);
+  const [button5, setButton5] = useState(false);
+  const [button6, setButton6] = useState(false);
+  const [button7, setButton7] = useState(false);
+  const [building, updateBuilding] = useState("");
+
+  function handleClick() {
+    setSelected(!selected);
+  }
+
+  function FilterButton() {
+    return (
+      <button
+        onClick={handleClick}
+        className={selected ? "button-toggled" : "button-default"}
+      >
+        Advanced
+      </button>
+    );
+  }
+
+  function QueryManager() {
```

Comment:
I would actually make query manager its own component and just use it in the homepage.